app/controllers/api/v1/tokens_controller.rb

```diff
@@ -82,7 +82,16 @@ def generate
         context: { bearer: current_bearer },
         with: TokenPolicy
 
+      # TODO(ezekg) make default session expiry configurable
+      session = token.sessions.build(
+        expiry: token.expiry.presence || 1.week.from_now,
+        user_agent: request.user_agent,
+        ip: request.remote_ip,
+      )
+
       if token.save
+        cookies.encrypted[:session_id] = { value: session.id, httponly: true, secure: true, same_site: :none, expires: session.expiry, domain: Keygen::DOMAIN }
```

Comment:
What if we're generating a token on behalf of another bearer, but don't want to clobber our session? E.g. an admin generating an environment token for an integration. I guess use the object's tokens relationship instead of the root? E.g. `POST /v1/environments/<environment>/tokens` instead of `/v1/tokens`?

Reply:
Or we could start a session only on `:password` authn?

Reply:
I say we leave as-is so that impersonation is possible. Cookies/credentials can always be ignored if this isn't desired, or the requestor can use one of the other token relationships instead.